src/bin/lib/subcommands.c/commandline.c

```diff
@@ -61,46 +68,47 @@ commandline_run(CommandLine *command, int argc, char **argv)
 
 	if (command->run != NULL)
 	{
-		return command->run(argc, argv);
+		command->run(argc, argv);
+		return true;
 	}
-	else if (argc == 0)
+
+	if (argc == 0)
 	{
 		/*
 		 * We're at the end of the command line already, and command->run is
 		 * not set, which means we expected a subcommand to be used, but none
 		 * have been given by the user. Inform him.
 		 */
-		commandline_print_subcommands(command, stderr);
+		fprintf(stderr, "%s: expected a command\n", breadcrumb);
 	}
-	else
+	else if (command->subcommands != NULL)
 	{
-		if (command->subcommands != NULL)
-		{
-			CommandLine **subcommand = command->subcommands;
-
-			for (; *subcommand != NULL; subcommand++)
-			{
-				if (streq(argv[0], (*subcommand)->name))
-				{
-					commandline_add_breadcrumb(command, *subcommand);
+		CommandLine **subcommand = command->subcommands;
 
-					return commandline_run(*subcommand, argc, argv);
-				}
-			}
-
-			/* if we reach this code, we didn't find a subcommand */
+		for (; *subcommand != NULL; subcommand++)
+		{
+			if (streq(argv[0], (*subcommand)->name))
 			{
-				const char *breadcrumb =
-					command->breadcrumb == NULL ? argv0 : command->breadcrumb;
+				commandline_add_breadcrumb(command, *subcommand);
 
-				fprintf(stderr,
-						"%s: %s: unknown command\n", breadcrumb, argv[0]);
+				return commandline_run(*subcommand, argc, argv);
 			}
-
-			fprintf(stderr, "\n");
-			commandline_print_subcommands(command, stderr);
 		}
+
+		/* if we reach this code, we didn't find a subcommand */
+		fprintf(stderr, "%s: %s: unknown command\n", breadcrumb, argv[0]);
```

Comment:
The order in which things are displayed looks strange to me here...

Reply:
I didn't change this message, but do you mean it's better to have:
```
pg_autoctl create: unknown command: something-weird

Reply:
I can't remember now if the breadcrumb contains `argv0`... and that's needed to make the right decision. The tradition for unix tools is I believe to say `argv[0]: ...` for their error messages so that you know which tool it initiates from.

Reply:
So currently the ouptut for this command:
```
pg_autoctl create something-bad
```
Looks like this:
```
pg_autoctl create: something-bad: unknown command

Available commands:
  pg_autoctl create
    monitor    Initialize a pg_auto_failover monitor node
    postgres   Initialize a pg_auto_failover standalone postgres node
    formation  Create a new formation on the pg_auto_failover monitor

```

And for this command:
```
pg_autoctl something-bad
```
Looks like this:
```
pg_autoctl: something-bad: unknown command

Available commands:
  pg_autoctl
  + create   Create a pg_auto_failover node, or formation
  + drop     Drop a pg_auto_failover node, or formation
  + config   Manages the pg_autoctl configuration
  + show     Show pg_auto_failover information
  + enable   Enable a feature on a formation
  + disable  Disable a feature on a formation
  + perform  Perform an action orchestrated by the monitor
    run      Run the pg_autoctl service (monitor or keeper)
    stop     signal the pg_autoctl service for it to stop
    reload   signal the pg_autoctl for it to reload its configuration
    help     print help message
    version  print pg_autoctl version
```


I think both look good, so the current values seem fine. If you have a command that displays something weird, I'm happy to fix it.